chain/network/src/time.rs

```diff
@@ -154,3 +194,28 @@ impl Default for FakeClock {
         Self::new(*FAKE_CLOCK_UTC_START)
     }
 }
+
+/// Interval equivalent to tokio::time::Interval with
+/// MissedTickBehavior::Skip.
+pub struct Interval {
+    next: time::Instant,
+    period: time::Duration,
+}
+
+impl Interval {
+    pub fn new(next: time::Instant, period: time::Duration) -> Self {
+        Self { next, period }
+    }
+
+    /// Cancel-safe.
+    pub async fn tick(&mut self, clock: &Clock) {
```

Comment:
I wonder if the `Interval` should perhaps just own the copy of the clock? 

Not sure, but I'd lean slightly towards the current API where the clock is passed in.